lib/services/suppressions-service.js

```diff
@@ -0,0 +1,270 @@
+                delete unused[relativeFilePath];
+            }
+        }
+
+        // Remove entries from unused that were not included in the results
+        for (const file in unused) {
+            if (!filtered.some(result => this.getRelativeFilePath(result.filePath) === file)) {
+                delete unused[file];
+            }
+        }
+
+        return {
+            results: filtered,
+            unused
+        };
+    }
+
+    /**
+     * Loads the suppressions file.
+     * @throws {Error} If the suppressions file cannot be parsed.
+     * @returns {Promise<SuppressedViolations>} The suppressions.
+     */
+    async load() {
+        try {
+            const data = await fs.promises.readFile(this.filePath, "utf8");
+
+            return JSON.parse(data);
+        } catch (err) {
+            if (err.code === "ENOENT") {
+                return {};
+            }
+            throw new Error(`Failed to parse suppressions file at ${this.filePath}`);
+        }
+    }
+
+    /**
+     * Updates the suppressions file.
+     * @param {SuppressedViolations} suppressions The suppressions to save.
+     * @returns {Promise<void>}
+     * @private
+     */
+    save(suppressions) {
+        return fs.promises.writeFile(
+            this.filePath,
+            JSON.stringify(suppressions, null, 2)
+        );
+    }
+
+    /**
+     * Counts the violations by rule, ignoring warnings.
+     * @param {LintMessage[]} messages The messages to count.
+     * @returns {Record<string, number>} The number of violations by rule.
+     */
+    static countViolationsByRule(messages) {
+        return messages.reduce((totals, message) => {
+            if (message.severity === 2) {
+                totals[message.ruleId] ??= { count: 0 };
+                totals[message.ruleId].count++;
+            }
+            return totals;
+        }, {});
+    }
+
+    /**
+     * Returns the relative path of a file to the current working directory.
+     * Always in POSIX format for consistency and interoperability.
+     * @param {string} filePath The file path.
+     * @returns {string} The relative file path.
+     */
+    getRelativeFilePath(filePath) {
+        return path
+            .relative(this.cwd, filePath)
+            .split(path.sep)
+            .join(path.posix.sep);
+    }
+
+    /**
+     * Moves the messages matching the rule to `LintResult#suppressedMessages` and updates the stats.
+     * @param {LintResult} result The result to update.
+     * @param {string} ruleId The rule to suppress.
+     * @returns {void}
+     */
+    static suppressMessagesByRule(result, ruleId) {
+        const suppressedMessages = result.messages.filter(message => message.ruleId === ruleId);
+
+        result.suppressedMessages = result.suppressedMessages.concat(
+            suppressedMessages
+                .map(message => {
+                    message.suppressions = [{
+                        type: "file",
```

Comment:
```suggestion
                        kind: "file",
```

https://eslint.org/docs/latest/integrate/nodejs-api#-suppressedlintmessage-type